### PR TITLE
logcli: Allow negative filters when omitting stream selector with --stdin

### LINKED ISSUE #8565: logcli: allow negative filters when omitting stream selector with `--stdin`
**Describe the bug**
`logcli query --stdin` fails when omitting the `{source="logcli"}` stream selector and the query starts with a negative filter (`!`).

**To Reproduce**
```
// Works with |
$ echo foo | logcli query --stdin '|= "foo"'
2023/02/21 01:28:19 Common labels: {source="logcli"}
2023-02-21T01:28:19+01:00 {} foo

// Fails with !
$ echo foo | logcli query --stdin '!= "foo"'
2023/02/21 01:28:16 Query failed: parse error at line 0, col 1: syntax error: unexpected !=
```

### PR DESCRIPTION
**Pull request by ndrpnt**
**Which issue(s) this PR fixes**:
Fixes #8565

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] `CHANGELOG.md` updated

### CODE REVIEW

**Review by chaudum — COMMENTED** [review]
LGTM Thanks for the fix!

Can you please also add a changelog entry beneath the `LogCLI` section?

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,13 +14,18 @@
...
 
 ##### Enhancements
+
 * [8474](https://github.com/grafana/loki/pull/8787) **andriikushch**: Promtail: Add a new target for the Azure Event Hubs
 
+#### LogCLI
+
+##### Fixes
+
+* [8566](https://github.com/grafana/loki/pull/8566) **ndrpnt**: Allow queries to start with negative filters (`!=` and `!~`) when omitting stream selector with `--stdin` flag
```
Comment (jeschkies) [inline CHANGELOG.md:27]: I'm curious what it has to do with the stdin flag.

**Review by jeschkies — APPROVED** [review]
Now I get it. Thanks :slightly_smiling_face:

**Comment by chaudum** [comment]
@ndrpnt Sorry for the long delay. DroneCI failing is unrelated. I think https://github.com/grafana/loki/pull/9154 should fix that, right @jeschkies ?

**Review by chaudum — APPROVED** [review]
lgtm. Thanks!

**Pull request merged**